### ISSUE TITLE
rename tenant to "project" 

### ISSUE THREAD

**Issue by msterin** [vmware/docker-volume-vsphere#982]
"tenant" is perceived as something related to a group of users and is confusing on out context. We need to emphasize that this is a way to limit access and privileges for a set of Vms , supposedly working on the same project. So the suggestion (from @pdhamdhere ) is to name it "project" , instead of "tenant".

this would include : 
* admin CLI
* documentation 
* error and help messages , if any, in esx_service and guest-side code
* auth_db tabes names (optional , can stay as it is)

//CC @lipingxue @shaominchen @pdhamdhere

**Comment by shaominchen** [comment]
I'm not sure if the name "project" is really better than "tenant" :-) This name may cause some confusion in technical communications - what does it mean to "create a project"? It's kind of ambiguous to me.

Personally I don't have strong concern about the name "tenant". But if most people think it's inappropriate, I'd prefer "role" or something else.

**Comment by shuklanirdesh82** [comment]
from https://github.com/vmware/docker-volume-vsphere/blob/master/docs/features/tenancy.md
"vSphere Administrator can define group of one or more Docker Host (VM) as Tenant"

e.g.
```
vmdkops_admin.py tenant create --name=tenant1 --vm-list=photon4
```

How about having it as ```vm-group```? simply it is a logical unit of dockerHosts to have privileges for datastore.

my two-cents

**Comment by govint** [comment]
"vm-group" opens up the implementation bellow the docker host, "project"
will match well with the photon platform (they have tenants and projects).
HOw about "host-group"

**Comment by lipingxue** [comment]
@govint what does "projects" and "tenants" mean in photon platform? I think "host-group" sounds a good name. Any comments @pdhamdhere @msterin

**Comment by msterin** [comment]
I agree with @govint here. `vm-group` is the best IMO. `host` ia easy to mistake for ESX host.

**Comment by tusharnt** [comment]
Fixed by #1021
